wordpress-shortcodes/src/main/java/org/wordpress/aztec/plugins/shortcodes/extensions/CaptionExtensions.kt

```diff
@@ -0,0 +1,51 @@
+package org.wordpress.aztec.plugins.shortcodes.extensions
+
+import android.text.Spanned
+import org.wordpress.aztec.AztecAttributes
+import org.wordpress.aztec.AztecText
+import org.wordpress.aztec.plugins.shortcodes.CaptionShortcodePlugin
+import org.wordpress.aztec.plugins.shortcodes.spans.CaptionShortcodeSpan
+import org.wordpress.aztec.spans.AztecImageSpan
+import org.wordpress.aztec.spans.IAztecNestable
+import org.wordpress.aztec.util.SpanWrapper
+
+fun AztecText.getImageCaption(attributePredicate: AztecText.AttributePredicate): String {
+    return this.text
+            .getSpans(0, this.text.length, AztecImageSpan::class.java)
+            .first {
+                attributePredicate.matches(it.attributes)
```

Comment:
This and same part in `setImageCaption` will throw an exception if nothing is found. Maybe we should just ignore cases where we can't find image span?

Reply:
Oh yeah, you're right! :+1: